proposals/0003-health-checks.md

```diff
@@ -0,0 +1,103 @@
+# Service Healthchecks
+
+* Proposal: [MP-0003](0003-health-checks.md)
+* Authors: [Clement Escoffier](https://github.com/cescoffier), [Heiko Rupp](https://github.com/pilhuhn), [Heiko Braun] (https://github.com/heiko-braun)
+
+* Status: **Awaiting review**
+
+*During the review process, add the following fields as needed:*
+
+* Decision Notes: [Discussion thread topic covering the  Rationale](https://groups.google.com/forum/#!forum/microprofile), [Discussion thread topic with additional Commentary](https://groups.google.com/forum/#!forum/microprofile)
+
+## Introduction
+
+This document describes a protocol (wireformat, semantics and possible forms of interactions) between system components that need to determine the “liveliness” of computing nodes in a bigger architecture.
+
+
+Mailinglist thread: [Discussion thread topic for that proposal](https://groups.google.com/forum/#!topic/microprofile/jIZAKiu76ys)
+
+## Motivation
+
+Health checks are used to probe the state of a computing node from another other machine (i.e. kubernetes service controller) with the primary target being cloud infrastructure environments where automated processes maintain the state of computing nodes. In this scenario health checks are used to determine if a computing node needs to be discarded (terminated, shutdown) and eventually replaced by another (healthy) instance.
```

Comment:
thanks for the clarification @heiko-braun. I agree with you on why we shouldn't be overloading the services. love the simplistic approach 👍